src/components/Filters/MoreOrLess/MoreOrLess.spec.js

```diff
@@ -0,0 +1,44 @@
+import { testRender as render, screen } from '../../../testUtils/test-utils';
+import userEvent from '@testing-library/user-event';
+import MoreOrLess from './MoreOrLess';
+import AggregationItem from '../AggregationItem/AggregationItem';
+
+const fixture = [
+  { key: 'alpha', doc_count: 99 },
+  { key: 'beta', doc_count: 99 },
+  { key: 'gamma', doc_count: 99 },
+  { key: 'delta', doc_count: 99 },
+  { key: 'epsilon', doc_count: 99 },
+  { key: 'zeta', doc_count: 99 },
+  { key: 'eta', doc_count: 99 },
+  { key: 'theta', doc_count: 99 },
+];
+
+const renderComponent = (props) => {
+  return render(<MoreOrLess {...props} />);
+};
+
+describe('component:MoreOrLess', () => {
+  const user = userEvent.setup({ delay: null });
+  const props = {
+    hasMore: true,
+    listComponent: AggregationItem,
+    listComponentProps: { fieldName: 'myfield' },
+    options: fixture,
+  };
+
+  test('matches - Show NN less', () => {
+    renderComponent(props);
+    expect(
+      screen.getByRole('button', { name: /Show 3 less/ }),
+    ).toBeInTheDocument();
+  });
```

Comment:
You can get rid of this test case since the one below it already covers this functionality.

Reply:
Unit tests have been updated to show initial state and toggle state for presence/non-presence of remaining results